src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/GitNativeRepoCloner.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+/// <summary>
+/// Class that clones repos by calling out to the native git executable.
+/// Does not uses Libgit2Sharp which is quite memory hungry when dealing with large repos.
+/// </summary>
+public class GitNativeRepoCloner : IGitRepoCloner
+{
+    private const string GitAuthUser = "dn-bot";
+
+    private readonly IProcessManager _processManager;
+    private readonly ILogger _logger;
+    private readonly string? _token;
+
+    public GitNativeRepoCloner(IProcessManager processManager, ILogger logger, string? token)
+    {
+        _processManager = processManager;
+        _logger = logger;
+        _token = token;
+    }
+
+    public Task CloneAsync(string repoUri, string? commit, string targetDirectory, bool checkoutSubmodules, string? gitDirectory)
+        => CloneAsync(
+            repoUri,
+            commit,
+            targetDirectory,
+            checkoutSubmodules ? CheckoutType.CheckoutWithSubmodules : CheckoutType.CheckoutWithoutSubmodules,
+            gitDirectory);
+
+    public Task CloneAsync(string repoUri, string targetDirectory, string? gitDirectory)
+        => CloneAsync(repoUri, null, targetDirectory, CheckoutType.NoCheckout, gitDirectory);
+
+    private async Task CloneAsync(
+        string repoUri,
+        string? commit,
+        string targetDirectory,
+        CheckoutType checkoutType,
+        string? gitDirectory)
+    {
+        _logger.LogInformation("Cloning {repoUri} to {targetDirectory}", repoUri, targetDirectory);
+
+        var args = new List<string>();
+
+        if (gitDirectory != null)
+        {
+            args.Add("--git-dir");
+            args.Add(gitDirectory);
+        }
+
+        args.Add("clone");
+        args.Add("-q");
+
+        if (checkoutType == CheckoutType.NoCheckout || commit != null)
+        {
+            args.Add("--no-checkout");
+        }
+        else if (checkoutType == CheckoutType.CheckoutWithSubmodules)
+        {
+            args.Add("--recurse-submodules");
+        }
+
+        args.Add("--");
+        args.Add(AddToken(repoUri));
```

Comment:
Even though some strings are redacted, in the logs, if I run this locally does the "origin" remote end up with a PAT in it?

Reply:
Ah, I see. Yes it would get left behind in the git config and running `git remote -v` would show it to you.
I will check if we can utilize a HTTP header instead

Reply:
Okay, it works, the logged command is this:

```
git -c http.extraheader=Authorization: Basic [REDACTED] clone -q --no-checkout https://dev.azure.com/dnceng/internal/_git/dotnet-installer /data/tmp/installer2
```